### PR TITLE
MINOR: Move MetadataQuorumCommand from `core` to `tools`

### PR DESCRIPTION
**Pull request by ijuma**
`core` should only be  used for legacy cli tools and tools that require
access to `core` classes instead of communicating via the kafka protocol
(typically by using the client classes).

Summary of changes:
1. Convert the command implementation and tests to Java and move it to
    the `tools` module.
2. Introduce mechanism to capture stdout and stderr from tests.
3. Change `kafka-metadata-quorum.sh` to point to the new command class.
4. Adjusted the test classpath of the `tools` module so that it supports tests
    that rely on the `@ClusterTests` annotation.
5. Improved error handling when an exception different from `TerseFailure` is
    thrown.
6. Changed `ToolsUtils` to avoid usage of arrays in favor of `List`.

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

**Comment by divijvaidya** [comment]
> core should only be used for legacy cli tools and tools that don't require
access to core classes instead of communicating via the kafka protocol
(typically by using the client classes).

It would be very useful to new contributors if you could add this information (about package structure/ownership) to https://kafka.apache.org/coding-guide or a package info file.

**Comment by ijuma** [comment]
> It would be very useful to new contributors if you could add this information (about package structure/ownership) to https://kafka.apache.org/coding-guide or a package info file.

Will do.

File: bin/kafka-metadata-quorum.sh [hunk 1]
```diff
@@ -14,4 +14,4 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-exec $(dirname $0)/kafka-run-class.sh kafka.admin.MetadataQuorumCommand "$@"
```
Comment (dengziming) [inline bin/kafka-metadata-quorum.sh:17]: There is a `kafka-metatada-quorum.bat` in the windows folder.

**Review by dengziming — APPROVED** [review]
LGTM, left one comment.

File: bin/kafka-metadata-quorum.sh (hunk 1 above)
Comment (ijuma) [reply bin/kafka-metadata-quorum.sh:17]: Good catch! Fixed.

**Comment by ijuma** [comment]
Failures are unrelated. Merging.

**Comment by ijuma** [comment]
By the way, I filed https://issues.apache.org/jira/browse/KAFKA-14525 for moving other cli tools to the `tools` module.

**Pull request merged**